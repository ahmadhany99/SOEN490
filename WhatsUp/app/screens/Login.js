/* eslint-disable react/no-unescaped-entities */
import React from "react";
import { useState } from "react";
import { StyleSheet, Text, View, Image, Alert } from "react-native";
import AppButton from "../components/AppButton";
import AppTextInput from "../components/AppTextInput";
import Links from "../components/Links";
import Screen from "../components/Screen";
import TitleHeaders from "../components/TitleHeaders";
import logo from "../Images/w3.png";
import { auth, db } from "../firebase";
import { collection, getDocs, query, where } from "firebase/firestore";
import { signInWithEmailAndPassword } from "firebase/auth";
import { useNavigation } from "@react-navigation/native";
import colors from "../config/colors";
import { SimpleLineIcons } from "@expo/vector-icons";

export default function Login() {
  const navigation = useNavigation();
  const [email, setEmail] = useState("jasmine@organizer.com");
  const [password, setPassword] = useState("capstone123");
  const [isOrganizer] = useState(false);
  const handleLogin = async () => {
    await signInWithEmailAndPassword(auth, email, password)
      .then(async () => {
        const q = query(collection(db, "users"), where("email", "==", email));
        const querySnapshot = await getDocs(q);
        if (querySnapshot != null) {
          querySnapshot.forEach((doc) => {
            if (isOrganizer == doc.data().isOrganizer) {
              navigation.navigate("Attendee");
              setPassword("");
            } else {
              navigation.navigate("Organizer");
              setPassword("");
            }
          });
        }
      })
      .catch(() => {
        Alert.alert("Try again", "Invalid email or password.");
      });
  };
  const SignupPressed = async () => {
    navigation.navigate("SignUp");
  };

  const ResetPassword = () => {
    navigation.navigate("ResetPassword");
  };
  return (
    <Screen
      resizeMode='cover'
      backgroundImage='https://cdn.dribbble.com/users/479289/screenshots/4521207/media/c3e3bd246bbff91101a54e69daa8b1f0.gif'
      style={{ paddingHorizontal: "10%" }}
    >
      <Image
        source={logo}
        style={{ marginTop: 18, width: 166, height: 212, alignSelf: "center" }}
      />
      <TitleHeaders
        style={{ fontSize: 38, marginTop: 20, marginBottom: 5 }}
        title='Login'
        isTitle={true}
      />
      <AppTextInput
        placeholder='Email'
        keyboardType='email-address'
        onChangeText={(text) => setEmail(text.toLowerCase())}
        value={email}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <AppTextInput
        placeholder='Password'
        secureTextEntry
        onChangeText={(text) => setPassword(text)}
        value={password}
        style={{
          shadowColor: "black", // IOS
          shadowOffset: { height: 1, width: 1 }, // IOS
          shadowOpacity: 0.2, // IOS
          shadowRadius: 2, //IOS
          elevation: 4, // Android
        }}
      />
      <View style={styles.organizertwo}>
        <Links
          style={styles.link}
          link='Forgot Password?'
          onPress={ResetPassword}
        />
      </View>
      <AppButton title='Login' onPress={handleLogin} />
      <View style={{ flexDirection: "row", justifyContent: "center" }}>
        <Text style={styles.text}>Don't have an account? </Text>
        <Links style={styles.link} link='Sign up' onPress={SignupPressed} />
      </View>
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          paddingVertical: 15,
        }}
      >
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
        <View>
          <Text
            style={{
              width: 105,
              textAlign: "center",
              color: "darkgray",
              fontSize: 14,
            }}
          >
            Or login with
          </Text>
        </View>
        <View style={{ flex: 1, height: 1, backgroundColor: "lightgrey" }} />
      </View>
      <View style={styles.rowContainer}>
        <SimpleLineIcons
          name='social-facebook'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            alignSelf: "center",
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='facebook'
        />
        <SimpleLineIcons
          name='social-twitter'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingHorizontal: 40,
            alignSelf: "center",
          }}
          onPress={() => console.log("Facebook")}
          testID='twitter'
        />
        <SimpleLineIcons
          name='social-google'
          size={30}
          style={{
            color: colors.secondary,
            fontSize: 30,
            paddingVertical: 10,
          }}
          onPress={() => console.log("Facebook")}
          testID='google'
        />
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  organizer: {
    alignItems: "flex-start",
    width: "50%",
  },
  organizertwo: {
    alignItems: "flex-end",
    width: "98%",
  },
  rowContainer: {
    flexDirection: "row",
    alignSelf: "center",
  },
  roundshape: {
    backgroundColor: "grey",
    height: 65,
    width: 65,
    justifyContent: "center",
    alignSelf: "center",
    alignContent: "center",
    borderRadius: 45,
  },
  text: {
    color: colors.darkerGrey,
    fontSize: 16,
  },
  text2: {
    fontSize: 16,
    color: colors.darkGrey,
  },
});
