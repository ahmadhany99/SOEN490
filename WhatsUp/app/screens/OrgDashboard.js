/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableOpacity,
  FlatList,
} from "react-native";
import Screen from "../components/Screen";
import colors from "../config/colors";
import { Ionicons } from "@expo/vector-icons";
import EventBanner from "../components/EventBanner";
import SearchBar from "react-native-dynamic-search-bar";
import EventImage from "../assets/stringio.jpg";
import { useNavigation } from "@react-navigation/native";
import { useAuthState } from "react-firebase-hooks/auth";
import { auth, db } from "../firebase";
import {
  collection,
  getDocs,
  query,
  where,
} from "firebase/firestore";

function OrganizerDashboardScreen() {
  const navigation = useNavigation();

  const [userName, setUserName] = useState("");
  const [user] = useAuthState(auth);

  const getName = async () => {
    const q = query(collection(db, "users"), where("email", "==", user.email));
    const querySnapshot = await getDocs(q);
    if (querySnapshot != null) {
      querySnapshot.forEach((doc) => {
        setUserName(doc.data().firstName);
      });
    }
  };

  getName();

  const [date, setDate] = useState(null);
  useEffect(() => {
    let d = new Date();
    const months = [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ];
    let today = "Today's " + months[d.getMonth()] + " " + d.getDate();
    setDate(today);
    setMasterData(events);
    setPreviousData(eventsPast);
  }, []);

  const events = [
    {
      image: "../assets/Logos/w1.png",
      title: "Fashion Week",
      organizer: "Lasalle College",
      date: "May 21, 2020",
    },
    {
      image: { EventImage },
      title: "Orientation Week",
      organizer: "ETS",
      date: "May 21, 2022",
    },
    {
      image: { EventImage },
      title: "FROSH",
      organizer: "Concordia Universityyyyyy",
      date: "May 21, 2023",
    },
    {
      image: { EventImage },
      title: "Film Festival",
      organizer: "Cinema",
      date: "May 21, 2024",
    },
    {
      image: { EventImage },
      title: "Anime Film Festival",
      organizer: "Cinema",
      date: "June 24, 2024",
    },
  ];

  const eventsPast = [
    {
      image: "../assets/Logos/w1.png",
      title: "Sports Weekend",
      organizer: "Concordia University",
      date: "May 21, 2022",
    },
    {
      image: { EventImage },
      title: "Music Festival",
      organizer: "Concordia University",
      date: "May 21, 2022",
    },
    {
      image: { EventImage },
      title: "Film Festival",
      organizer: "Cinema",
      date: "May 21, 2024",
    },
    {
      image: { EventImage },
      title: "Jazz Festival",
      organizer: "Concordia University",
      date: "May 21, 2022",
    },
    {
      image: { EventImage },
      title: "F1 ",
      organizer: "Concordia University",
      date: "May 21, 2022",
    },
  ];

  const [displayedEvent, setDisplayedEvents] = useState(true);
  const [search, setSearch] = useState("");
  const [masterData, setMasterData] = useState([]);
  const [previousData, setPreviousData] = useState([]);
  const [filteredData, setFilteredData] = useState("");
  const [filteredOrgData, setFilteredOrgData] = useState("");

  const ItemView = ({ item }) => {
    return (
      <EventBanner
        image={EventImage}
        title={item.title}
        organizer={item.organizer}
        date={item.date}
      />
    );
  };

  const searchFilter = (text) => {
    if (text && displayedEvent) {
      const newData = masterData.filter((item) => {
        const itemData = item.title
          ? item.title.toUpperCase()
          : "".toUpperCase();
        const textData = text.toUpperCase();
        return itemData.indexOf(textData) > -1;
      });

      const orgSearch = masterData.filter((item) => {
        const itemData = item.organizer
          ? item.organizer.toUpperCase()
          : "".toUpperCase();
        const textData = text.toUpperCase();
        return itemData.indexOf(textData) > -1;
      });

      setFilteredOrgData(orgSearch);
      setFilteredData(newData);
      console.log(filteredData);
      setSearch(text);
    } else if (text && !displayedEvent) {
      const newData = previousData.filter((item) => {
        const itemData = item.title
          ? item.title.toUpperCase()
          : "".toUpperCase();
        const textData = text.toUpperCase();
        return itemData.indexOf(textData) > -1;
      });

      const orgSearch = previousData.filter((item) => {
        const itemData = item.organizer
          ? item.organizer.toUpperCase()
          : "".toUpperCase();
        const textData = text.toUpperCase();
        return itemData.indexOf(textData) > -1;
      });

      setFilteredOrgData(orgSearch);
      setFilteredData(newData);
      console.log(filteredData);
      setSearch(text);
    } else {
      displayedEvent
        ? setFilteredData(masterData)
        : setFilteredData(previousData);
      setSearch(text);
    }
  };

  const toggleDisplay = () => {
    setDisplayedEvents({ displayedEvent: !displayedEvent });
  };

  var tabs;
  var showEvents;
  if (displayedEvent) {
    tabs = (
      <>
        <TouchableOpacity
          title="Show Form 1"
          onPress={() => setDisplayedEvents(true)}
          style={styles.upcoming}
        >
          <Text style={{ fontSize: 16, fontWeight: "bold" }}>Upcoming</Text>
        </TouchableOpacity>
        <TouchableOpacity
          title="Show Form 2"
          onPress={() => setDisplayedEvents(false)}
          style={styles.previous}
        >
          <Text style={{ fontSize: 16, fontWeight: "bold" }}>Previous</Text>
        </TouchableOpacity>
      </>
    );
    showEvents = (
      <>
        <FlatList
          data={filteredData ? filteredData : events}
          renderItem={ItemView}
          style={{}}
        />
        <FlatList
          data={filteredOrgData ? filteredOrgData : []}
          renderItem={ItemView}
        />
      </>
    );
  } else {
    tabs = (
      <>
        <TouchableOpacity
          title="Show Form 1"
          onPress={() => setDisplayedEvents(true)}
          style={styles.previous}
        >
          <Text style={{ fontSize: 16, fontWeight: "bold" }}>Upcoming</Text>
        </TouchableOpacity>
        <TouchableOpacity
          title="Show Form 2"
          onPress={() => setDisplayedEvents(false)}
          style={styles.upcoming}
        >
          <Text style={{ fontSize: 16, fontWeight: "bold" }}>Previous</Text>
        </TouchableOpacity>
      </>
    );
    showEvents = (
      <>
        <FlatList
          data={filteredData ? filteredData : eventsPast}
          renderItem={ItemView}
        />
        <FlatList
          data={filteredOrgData ? filteredOrgData : []}
          renderItem={ItemView}
        />
      </>
    );
  }
  return (
    <Screen style={{ padding: 20, marginTop: 10 }}>
      <View>
        <View style={styles.header}>
          <View style={styles.headerContent}>
            <Text style={{ color: colors.darkGrey }}>
              <Text style={styles.paragraph}>{date}</Text>{" "}
            </Text>
            <Text style={{ fontWeight: "bold", fontSize: 28 }}>
              Welcome, {userName}!
            </Text>
          </View>
          <TouchableOpacity>
            <Ionicons
              name="add-circle-outline"
              size={40}
              color={colors.primary}
              onPress={() => navigation.navigate("NewEvent")}
            />
          </TouchableOpacity>
        </View>

        <View style={styles.searchBar}>
          <SearchBar
            style={{ width: "85%" }}
            placeholder="Search for..."
            onChangeText={(text) => {
              searchFilter(text);
            }}
            // Icon={{ visible: searchText.length > 0 }}
            />
          <TouchableOpacity style={styles.filter}>
            <Ionicons name="ios-filter" size={24} color={colors.primary} />
          </TouchableOpacity>
        </View>
        <Text style={styles.eventTitle}>Your Events</Text>
      </View>
      {showEvents}
        <View style={styles.eventTabs}>{tabs}</View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  organizer: {
    alignItems: "flex-start",
    width: "50%",
  },
  organizertwo: {
    alignItems: "flex-end",
    width: "50%",
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  eventTitle: {
    fontWeight: "bold",
    fontSize: 18,
    marginTop: 16,
    marginBottom: 10,
  },
  searchBar: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginTop: 18,
  },
  filter: {
    justifyContent: "center",
  },
  eventTabs: {
    flexDirection: "row",
    justifyContent: "space-around",
    marginTop: 2,
  },
  upcoming: {
    marginTop: 4,
    borderBottomWidth: 2,
    borderBottomColor: colors.primary,
    paddingBottom: 10,
    fontWeight: "bold",
  },
  previous: {
    marginTop: 4,
    paddingBottom: 10,
  },
});

export default OrganizerDashboardScreen;
