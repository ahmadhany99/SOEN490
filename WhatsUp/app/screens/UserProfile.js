/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import { View, Image, StyleSheet, ScrollView, Alert } from "react-native";
import Screen from "../components/Screen";
import ScreenTitle from "../components/ScreenTitle";
import { Ionicons } from "@expo/vector-icons";
import logo from "../Images/w3.png";
import ImgOrgBottom from "../components/ImgOrgBottom";
import ScreenSubtitle from "../components/ScreenSubtitle";
import Links from "../components/Links";
import colors from "../config/colors";
import AppModal from "../components/AppModal";
import BackBtn from "../components/BackBtn";
import AppTextInput from "../components/AppTextInput";
import AppButton from "../components/AppButton";
import { useAuthState } from "react-firebase-hooks/auth";
import { auth, db } from "../firebase";
import { sendPasswordResetEmail, updateEmail } from "firebase/auth";
import {
  collection,
  getDocs,
  query,
  where,
  doc,
  updateDoc,
} from "firebase/firestore";
import { fetchSignInMethodsForEmail } from "firebase/auth";

function UserProfile() {
  const [modalVisibleName, setModalVisibleName] = useState(false);
  const [modalVisibleEmail, setModalVisibleEmail] = useState(false);
  const [newName, setNewName] = useState("");
  const [newEmail, setNewEmail] = useState("");
  const [docKey, setDocKey] = useState("");
  const [userName, setUserName] = useState("");
  const [user] = useAuthState(auth);

  const getName = async () => {
    const q = query(collection(db, "users"), where("email", "==", user.email));
    const querySnapshot = await getDocs(q);
    if (querySnapshot != null) {
      querySnapshot.forEach((doc) => {
        setUserName(doc.data().firstName + " " + doc.data().lastName);
      });
    }
  };

  const getKey = async () => {
    const q = query(collection(db, "users"), where("email", "==", user.email));
    const querySnapshot = await getDocs(q);
    if (querySnapshot != null) {
      querySnapshot.forEach((doc) => {
        setDocKey(doc.id);
      });
    }
  };

  getName();
  getKey();

  const updateFirestoreEmail = async () => {
  const docRef = doc(db, "users", docKey);
  const data = {
    email: newEmail,
  };
  await updateDoc(docRef, data)
    .then(() => {
    })
    .catch((error) => {
      console.log(error);
    });
  }

  const updateEmailForCurrentUser = () => {
    updateEmail(user, newEmail)
      .then(() => {
        // Email updated!
        // ...
      })
      .catch((error) => {
        // An error occurred
        // ...
      });
  };

  const sendResetEmail = () => {
    sendPasswordResetEmail(auth, user.email)
      .then(() => {
        Alert.alert(
          "Email sent sucessfully.",
          "Check your emails to reset the password."
        );
      })
      .catch(() => {});
  };

  const checkIfEmailExists = () => {
    fetchSignInMethodsForEmail(auth, user.email)
      .then((result) => {
        if (result === undefined || result.length == 0) {
          Alert.alert("We couldn't find an account with that email address.");
        } else {
          //If email exists, send the link
          sendResetEmail();
        }
      })
      .catch((error) => {
        console.log(error.message);
        Alert.alert(error.message);
      });
  };

  const checkIfEmailAlreadyExists = () => {
    fetchSignInMethodsForEmail(auth, newEmail)
      .then((result) => {
        if (result === undefined || result.length == 0) {
          Alert.alert("Email was successfully updated.");
          updateEmailForCurrentUser(newEmail);
        } else {
          Alert.alert("The email already exists.");
        }
      })
      .catch((error) => {
        console.log(error.message);
        Alert.alert(error.message);
      });
  };

  return (
    <Screen style={{ padding: 20, backgroundColor: "#F5F5F5" }}>
      <View style={{ left: "1.5%", marginTop: "5%" }}>
        <ScreenTitle title="Profile" style={{ marginBottom: "2%" }} />

        <Image
          source={logo}
          style={{
            width: 166,
            height: 212,
            alignSelf: "center",
            marginTop: "5%",
          }}
        />
      </View>
      <View style={{ flexDirection: "row" }}>
        <ScreenTitle
          title="Name"
          style={{ marginBottom: "2%", fontSize: 20, marginTop: "10%" }}
        />
        {/* code if we ever want to edit the name: */}
        {/* <Ionicons
          name="pencil"
          onPress={() => setModalVisibleName(true)}
          size={20}
          color={colors.primary}
          style={{
            marginBottom: "2%",
            marginTop: "10%",
            alignSelf: "flex-end",
            marginLeft: "auto",
          }}
        /> */}
      </View>
      <View>
        <ScreenSubtitle
          subtitle={userName}
          style={{
            marginBottom: "2%",
            marginTop: "0.5%",
            marginLeft: "1%",
            fontSize: 16,
          }}
        />
      </View>
      <View style={{ flexDirection: "row" }}>
        <ScreenTitle
          title="Email"
          style={{ marginBottom: "2%", fontSize: 20, marginTop: "5%" }}
        />
        <Ionicons
          name="pencil"
          onPress={() => setModalVisibleEmail(true)}
          size={20}
          color={colors.primary}
          style={{
            marginBottom: "2%",
            marginTop: "5%",
            alignSelf: "flex-end",
            marginLeft: "auto",
          }}
        />
      </View>
      <View>
        <ScreenSubtitle
          subtitle={user.email}
          style={{
            marginBottom: "2%",
            marginTop: "0.5%",
            marginLeft: "1%",
            fontSize: 16,
          }}
        />
        <ScreenTitle
          title="Password"
          style={{ marginBottom: "2%", fontSize: 20, marginTop: "5%" }}
        />
        <Links
          style={{ marginLeft: "1%", marginTop: "0.5%", fontSize: 16
           }}
          link="Change Password"
          onPress={checkIfEmailExists}
        />
      </View>
      <ImgOrgBottom resizeMode="contain" />
      <AppModal
        animationType="slide"
        transparent={true}
        visible={modalVisibleName}
        onRequestClose={() => {
          setModalVisibleName(!modalVisible);
        }}
      >
        <View style={styles.modalView}>
          <BackBtn
            style={styles.backModal}
            onPress={() => setModalVisibleName(!modalVisibleName)}
          />
          <View style={styles.inputView}>
            <ScrollView
              keyboardDismissMode="interactive"
              style={{ width: "100%" }}
            >
              <ScreenTitle
                style={{
                  alignSelf: "center",
                  fontSize: 22,
                  marginBottom: "5%",
                }}
                title={"Enter New Name:"}
              />
              <AppTextInput
                placeholder="Name"
                onChangeText={(currentName) => setNewName(currentName)}
              />
              <AppButton
                title="Submit"
                style={{ marginTop: 0 }}
                onPress={() => {
                  setModalVisibleName(!modalVisibleName);
                }}
              />
            </ScrollView>
          </View>
        </View>
      </AppModal>
      <AppModal
        animationType="slide"
        transparent={true}
        visible={modalVisibleEmail}
        onRequestClose={() => {
          setModalVisibleName(!modalVisibleName);
        }}
      >
        <View style={styles.modalView}>
          <BackBtn
            style={styles.backModal}
            onPress={() => setModalVisibleEmail(!modalVisibleEmail)}
          />
          <View style={styles.inputView}>
            <ScrollView
              keyboardDismissMode="interactive"
              style={{ width: "100%" }}
            >
              <ScreenTitle
                style={{
                  alignSelf: "center",
                  fontSize: 22,
                  marginBottom: "5%",
                }}
                title={"Enter New Email:"}
              />
              <AppTextInput
                placeholder="Email"
                onChangeText={(newEmail) => {
                  setNewEmail(newEmail);
                }}
              />
              <AppButton
                title="Submit"
                style={{ marginTop: 0 }}
                onPress={() => {
                  setModalVisibleEmail(!modalVisibleEmail);
                  checkIfEmailAlreadyExists();
                  updateFirestoreEmail(newEmail);
                }}
              />
            </ScrollView>
          </View>
        </View>
      </AppModal>
    </Screen>
  );
}

const styles = StyleSheet.create({
  modalView: {
    margin: 270,
    backgroundColor: "white",
    borderRadius: 35,
    padding: 20,
    paddingTop: 20,
    width: "86%",
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  inputView: {
    marginTop: 8,
    borderColor: colors.lightGrey,
    borderRadius: 7,
    width: "90%",
    alignSelf: "center",
    paddingTop: 20,
  },
  backModal: {
    backgroundColor: "black",
  },
});

export default UserProfile;
