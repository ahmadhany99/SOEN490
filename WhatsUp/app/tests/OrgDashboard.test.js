import "react-native";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
import OrgDashboard from "../screens/OrgDashboard";
import Login from "../screens/Login";
import UserProfile from "../screens/UserProfile";
import NewEvent from "../screens/OrgNewEvent";

jest.useFakeTimers();

it("Renders correctly organizer dashboard page.", () => {
  const tree = render(
    <NavigationContainer>
      <OrgDashboard />
    </NavigationContainer>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Login sucessfully with organizer account and render organizer dashboard page.", async () => {
  render(
    <NavigationContainer>
      <Login />
    </NavigationContainer>
  );
  await waitFor(() =>
    fireEvent.changeText(
      screen.getByPlaceholderText("Email"),
      "jasmine@organizer"
    )
  );
  await waitFor(() =>
    fireEvent.changeText(screen.getByPlaceholderText("Password"), "capstone123")
  );
  await waitFor(() => {
    fireEvent.press(screen.getAllByText("Login")[1]);
  });
  const org = (
    <NavigationContainer>
      <OrgDashboard />
    </NavigationContainer>
  );
  render(org);
});

test("Clicking on add Event Icon goes successfully to add new event page.", async () => {
  render(
    <NavigationContainer>
      <OrgDashboard />
    </NavigationContainer>
  );
  await waitFor(() => {
    fireEvent.press(screen.getByTestId("addEventButton"));
  });

  const newEvent = (
    <NavigationContainer>
      <NewEvent />
    </NavigationContainer>
  );
  render(newEvent);
});


test("Clicking on Profile Icon goes successfully to user profile page.", async () => {
  render(
    <NavigationContainer>
      <OrgDashboard />
    </NavigationContainer>
  );
  
  const temp = screen.queryAllByText("");
  await waitFor(() => {
    fireEvent.press(temp[5]);
  });

  const userProfile = (
    <NavigationContainer>
      <UserProfile />
    </NavigationContainer>
  );
  render(userProfile);
});


