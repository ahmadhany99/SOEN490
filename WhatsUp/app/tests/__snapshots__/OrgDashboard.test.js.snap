// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly organizer dashboard page. 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "marginTop": 10,
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "left": "2.5%",
        "marginTop": "5%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#747474",
              }
            }
          >
            <Text>
              Today's February 6
            </Text>
          </Text>
          <Text
            style={
              Object {
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }
          >
            Welcome, !
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 5,
                "marginTop": 15,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderBottomColor": "#00c0a4",
                  "borderBottomWidth": 2,
                  "fontWeight": "bold",
                  "marginLeft": "20%",
                  "marginTop": 5,
                  "opacity": 1,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Upcoming
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginLeft": "20%",
                  "marginTop": 5,
                  "opacity": 1,
                  "paddingBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Previous
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "elevation": 2,
              "flexDirection": "row",
              "opacity": 1,
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "size": 12,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 18,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#fdfdfd",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 40,
              "shadowColor": "#757575",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 8,
              "width": "90%",
            }
          }
        >
          <View>
            <Text />
          </View>
          <TextInput
            onChangeText={[Function]}
            placeholder="Search for..."
            style={
              Object {
                "color": "#19191a",
                "marginLeft": 12,
                "width": "80%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "elevation": 2,
              "flexDirection": "row",
              "marginLeft": "2%",
              "marginTop": "0.5%",
              "opacity": 1,
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "marginLeft": 2,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginTop": 16,
          }
        }
      >
        Your Events
      </Text>
    </View>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={Object {}}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
